src/settings.rs

```diff
@@ -8,17 +8,91 @@ pub struct ServiceConfig {
     pub host: String,
     pub http_port: u16,
     pub grpc_port: Option<u16>, // None means that gRPC is disabled
-    pub internal_grpc_port: Option<u16>, // None means that the internal gRPC is disabled
     pub max_request_size_mb: usize,
     pub max_workers: Option<usize>,
 }
 
+#[derive(Debug, Deserialize, Clone)]
+pub struct ClusterConfig {
+    pub enabled: bool, // TODO use with https://github.com/qdrant/qdrant/issues/511
+    #[serde(default)]
+    pub p2p: P2pConfig,
+    #[serde(default)]
+    pub consensus: ConsensusConfig,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct P2pConfig {
+    #[serde(default)]
+    pub p2p_port: Option<u16>,
+    #[serde(default = "default_timeout_ms")]
+    pub p2p_grpc_timeout_ms: u64,
+}
+
+impl Default for P2pConfig {
+    fn default() -> Self {
+        P2pConfig {
+            p2p_port: None,
+            p2p_grpc_timeout_ms: default_timeout_ms(),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct ConsensusConfig {
+    #[serde(default = "default_max_in_flight_messages")]
+    pub max_in_flight_messages: usize,
```

Comment:
Might be less confusing I think
```suggestion
    pub max_message_queue_size: usize,
```